proposals/NNNN-package-manager-product-definitions.md

```diff
@@ -110,41 +158,32 @@ Package(
 The definition of the `Product` type will be:
    
 ```
-public class Product {
-    public let name: String
-    public let targets: [String]
+public enum Product {
+    public class AbstractProduct {
+        public let name: String
+        public let targets: [String]
+        
+        public init(name: String, targets: [String])
+    }
```

Comment:
I wanted to have a common ancestor to indicate that there is an inheritance relationship between at least some kinds of Product.  Perhaps a protocol instead?